oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneAndOneComputeService.java

```diff
@@ -0,0 +1,102 @@
+import org.jclouds.compute.strategy.ResumeNodeStrategy;
+import org.jclouds.compute.strategy.SuspendNodeStrategy;
+import org.jclouds.domain.Credentials;
+import org.jclouds.domain.Location;
+import org.jclouds.scriptbuilder.functions.InitAdminAccess;
+
+@Singleton
+public class OneAndOneComputeService extends BaseComputeService {
+
+   private final CleanupResources cleanupResources;
+
+   @Inject
+   protected OneAndOneComputeService(ComputeServiceContext context, Map<String, Credentials> credentialStore,
+           @Memoized Supplier<Set<? extends Image>> images, @Memoized Supplier<Set<? extends Hardware>> sizes,
+           @Memoized Supplier<Set<? extends Location>> locations, ListNodesStrategy listNodesStrategy,
+           GetImageStrategy getImageStrategy, GetNodeMetadataStrategy getNodeMetadataStrategy,
+           CreateNodesInGroupThenAddToSet runNodesAndAddToSetStrategy, RebootNodeStrategy rebootNodeStrategy,
+           DestroyNodeStrategy destroyNodeStrategy, ResumeNodeStrategy startNodeStrategy,
+           SuspendNodeStrategy stopNodeStrategy, Provider<TemplateBuilder> templateBuilderProvider,
+           @Named("DEFAULT") Provider<TemplateOptions> templateOptionsProvider,
+           @Named(TIMEOUT_NODE_RUNNING) Predicate<AtomicReference<NodeMetadata>> nodeRunning,
+           @Named(TIMEOUT_NODE_TERMINATED) Predicate<AtomicReference<NodeMetadata>> nodeTerminated,
+           @Named(TIMEOUT_NODE_SUSPENDED) Predicate<AtomicReference<NodeMetadata>> nodeSuspended,
+           InitializeRunScriptOnNodeOrPlaceInBadMap.Factory initScriptRunnerFactory,
+           RunScriptOnNode.Factory runScriptOnNodeFactory, InitAdminAccess initAdminAccess,
+           PersistNodeCredentials persistNodeCredentials, Timeouts timeouts,
+           @Named(Constants.PROPERTY_USER_THREADS) ListeningExecutorService userExecutor,
+           CleanupResources cleanupResources, Optional<ImageExtension> imageExtension,
+           Optional<SecurityGroupExtension> securityGroupExtension) {
+      super(context, credentialStore, images, sizes, locations, listNodesStrategy, getImageStrategy,
+              getNodeMetadataStrategy, runNodesAndAddToSetStrategy, rebootNodeStrategy, destroyNodeStrategy,
+              startNodeStrategy, stopNodeStrategy, templateBuilderProvider, templateOptionsProvider, nodeRunning,
+              nodeTerminated, nodeSuspended, initScriptRunnerFactory, initAdminAccess, runScriptOnNodeFactory,
+              persistNodeCredentials, timeouts, userExecutor, imageExtension, securityGroupExtension);
+      this.cleanupResources = cleanupResources;
+   }
+
+   @Override
+   protected void cleanUpIncidentalResourcesOfDeadNodes(Set<? extends NodeMetadata> deadNodes) {
+      for (NodeMetadata deadNode : deadNodes) {
+         try {
+            cleanupResources.cleanupNode(deadNode.getId());
+         } catch (Exception ex) {
+            logger.warn(ex, "Error cleaning up resources for node %s", deadNode);
+         }
+      }
+   }
+}
```

Comment:
The compute service delete node already calls here, so this class shouldn't be necessary.